### ISSUE TITLE
[Accessibility Audit] Code Intel: Finding references for a definition

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#33510]
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1kA6aVOAgID_uPm-d6uEC1DG6WHjJ8fPT0UAXQcML_KQ/edit?usp=sharing). Use this for further context.

- Hover over an identifier.
- Click on the Find references button in the Code Intel Popover.
- Optional: Potentially increase height of the References panel. (Note: I wasn’t able to capture the cursor change in the screenshot. There is a very subtle shadow change too near the boundary, but it’s hard to see without zooming in.)
- Select repository and file of interest from the References panel.
- Click on code snippet to navigate to reference.

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @macraig - the "team/code-intelligence" label was applied to this issue.

**Comment by varungandhi-src** [comment]
(Moved from https://github.com/sourcegraph/sourcegraph/issues/33509#issuecomment-1119710219)

## Find references + New Ref panel (12)

- [X] If you click "Find Refs" it doesn't focus the reference panel and the focus appears to be on a random usage within the code view. Reference panel opening is not announced whether it's open already or not. https://github.com/sourcegraph/sourcegraph/issues/36568
- [X] Mix search and precise changes state but does not announce any change in the state. Check if we can add aria-live or something. https://github.com/sourcegraph/sourcegraph/issues/36570
- [X] On ref panel close button, it gives you options to exit the "web area" - if you go up to the parent you can't re-enter the ref panel. Is this expected? https://github.com/sourcegraph/sourcegraph/issues/36571
- [ ] Trying Ctrl + Option + Shift + ↓ to enter the web content from the ref panel header doesn't work, despite the suggestion from the screen reader (you get audible bell feedback that there's nothing to do).
- [ ] In ref panel, `Type to filter by filename` announces it's a text box, but would be nice to say how many files there are to filter, and how many matches there are as we type.
- [ ] Collapsible reads out strangely, as `Close <header>` instead of `<header>, click to collapse this content`, which would be much better. Additionally (on us) the text `(n displayed)` should probably be `(n matches)` or `(n occurrences)` to be more clear.
- [ ] Displaying per-repo match counts might be nice as well (we have them per-file and per type (def/ref/impl))
- [ ] Navigating down to a match will say "list item" but not read out the line number or the text
- [ ] Clicking on a match in the ref panel opens content to the right but the screen reader does not announce an action.
- [ ] It is time-consuming to navigate from the ref panel to the file content pane. @varungandhi-src's opinion: Maybe we should change the tab hierarchy so that we can quickly tab to it and back.
- [ ] Lack of clarity around number of repos for matches is a little disorienting. Maybe update text to be "References (5 occurrences *over 3 repositories*)". We should also probably indicate whether there is a load more option here.
- [X] Collapsible (and some other text elements it seems) allows right click. https://github.com/sourcegraph/sourcegraph/issues/35267
- [X] ~Tab breaks when closing a collapsible~ Could not reproduce again

**Comment by varungandhi-src** [comment]
Moved from https://github.com/sourcegraph/sourcegraph/issues/33511#issuecomment-1122703489

## Ref panel

- [X] In the button group, it mentions ~close~ "Type to filter by filename" twice. https://github.com/sourcegraph/sourcegraph/issues/36605
- [X] Maybe we need to have a prefix when you go thru the labels to know if its a repository/directory/file. https://github.com/sourcegraph/sourcegraph/issues/36606
- [X] you can get to heading but you cannot go into the nested buttons under the heading with control+option+right (already fixed as of June 4)
    - heading seems to be stuck in an island. Cannot go back to the level to the close button or list
- [X] Ctrl+option+down doesn’t seem to work on a list within the reference panel -- this doesn't seem necessary because Control + Option + ←/→ work fine.
- [X] says 1 displayed when filtered but its potentially more useful is to provide more context like how many more are there. https://github.com/sourcegraph/sourcegraph/issues/36580
- [X] ~Instead of saying button it should say clickable. I wonder if we are using buttons out of convenience but we should be using links instead.~ I think the behavior is correct (even though perhaps a little visually odd), based on whether it is a state change or navigating to somewhere else.

**Comment by varungandhi-src** [comment]
Closing this; I think we've covered all the issues we found related to Find references or the Reference panel.
